contents/ruby/Rails.md

```diff
@@ -0,0 +1,170 @@
+
+For example, in order to fetch yesterday's date in Python, one may have to write
+
+```python
+import datetime
+datetime.datetime.now() - datetime.timedelta(days = 1)
+```
+
+The same action in Rails is done using
+```ruby
+1.day.ago
+```
+
+Similarly, Rails also added a new method to access an array element. Instead of following the C convention of using square-brackets to access the second element in the array "elements" like
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, easy to work with, and, consequently, happiness-inducing.
+
+### Large and Active Community of Developers
+
+The Ruby language has a very passionate and extensive community of developers, and there is a vast array of libraries (called "Gems"), that can be used with your Rails projects. These libraries provide a lot of functionalities, and can be used to add flexibility to Rails projects when required. 
+
+Furthermore, as a result of its popularity, it is very likely that any problems you encounter with Rails will have a solution that you can find on Google or StackOverflow. For any new problems, you are also likely to receive a response very quickly, because the Ruby community prides itself for being welcoming and helpful. 
+
+However, Rails might not be best suited for all use cases, and there are certain areas where other frameworks might be more suitable. For instance:
+
+### Drawbacks
+
+#### Lack of Flexibility
+
+Although Rails is highly optimized for most types of web applications, there are inevitably some cases where following an MVC architecture isn't suitable. In those cases, it might be better for developers to use frameworks that allow for more flexibility. You may also wish to have a different directory structure, or give your routes different names, and find Rails' defaults very stifling. While there are ways for you to configure Rails to your needs, it might be less painful to avoid those conventions from the beginning.
+
+#### Performance
+
+While newer releases have combated this problem to a large degree, many of the old releases of Rails have had a reputation for being very bloated and slow to respond. Applications in other frameworks, or purely client-side applications with a minimal backend, tend to have a much lower delay between an action and a response. While it should be noted that the vast majority of delays in applications occur due to poor optimization, and not the inherent speed of the framework, Rails still remains more bulky and slower than many of its competitors. 
```

Comment:
This was pointed out by Marvin previously:
`combated` -> `combatted`

Reply:
Er.. he pointed out `combatted` -> `combated` , and as I understand it, they're both correct. Google docs doesn't flag either as an error, and this link says they're alternate forms. 
https://wikidiff.com/combatted/combated

Reply:
OH. Sorry my bad.